File: developer-service/src/main/java/com/appverse/developer_service/dto/DeveloperResponse.java
package com.appverse.developer_service.dto;

import java.time.Instant;

import com.appverse.developer_service.enums.DeveloperStatus;
import com.appverse.developer_service.enums.DeveloperType;

public record DeveloperResponse(
    String id,

    String username,
    String firstName,
    String lastName,
    String email,

    String website,
    String companyName,
    String bio,
    String logoUrl,
    String location,

    DeveloperStatus status,
    DeveloperType developerType,
    boolean isVerified,

    Instant createdAt
) {}

File: frontend/appverse/src/app/features/developer/models/developer-response.ts
export interface DeveloperResponse {
  id: string;
  firstName: string;
lastName: string;
  email: string;
  website?: string;
  companyName?: string;
  bio?: string;
  logoUrl?: string;
  location?: string;
  developerType: 'INDIVIDUAL' | 'ORGANIZATION';
  role: 'USER' | 'DEVELOPER' | 'ADMIN';
  status: 'ACTIVE' | 'INACTIVE' | 'PENDING_VERIFICATION';
  isVerified: boolean;
  createdAt: string;
}